include/z64bgcheck.h

```diff
@@ -0,0 +1,203 @@
+#ifndef _Z_BGCHECK_
+#define _Z_BGCHECK_
+
+struct GlobalContext;
+struct Actor;
+struct DynaPolyActor;
+
+#define SS_NULL 0xFFFF
+#define COLPOLY_NORM_FRAC (1.0f / 32767)
+#define COLPOLY_GET_NORM(n) ((n)*COLPOLY_NORM_FRAC)
+
+// bccFlags
+#define BGCHECK_CHECK_WALL 1
+#define BGCHECK_CHECK_FLOOR 2
+#define BGCHECK_CHECK_CEILING 4
+#define BGCHECK_CHECK_ONE_FACE 8
+#define BGCHECK_CHECK_DYNA 0x10
+
+// bciFlags
+#define BGCHECK_IGNORE_NONE 0
+#define BGCHECK_IGNORE_CEILING 1
+#define BGCHECK_IGNORE_WALL 2
+#define BGCHECK_IGNORE_FLOOR 4
+
+// xpFlags
+#define COLPOLY_IGNORE_NONE 0
+#define COLPOLY_IGNORE_CAMERA 1
+#define COLPOLY_IGNORE_ENTITY 2
+#define COLPOLY_IGNORE_PROJECTILES 4
+#define COLPOLY_SNORM(x) ((short)((x) * 32767))
+
+#define BG_ACTOR_MAX 50
+#define BGCHECK_SCENE BG_ACTOR_MAX
+#define BGCHECK_Y_MIN -32000.0f
+#define VIA_FLAG_TEST(vIA, flags) ((vIA) & (((flags)&7) << 13))
+#define VTX_INDEX(vI) ((vI)&0x1FFF)
+
+#define FUNC_80041EA4_RESPAWN 5
+#define FUNC_80041EA4_MOUNT_WALL 6
+#define FUNC_80041EA4_STOP 8
+#define FUNC_80041EA4_VOID_OUT 12
+
+#define WATERBOX_ROOM(p) ((p >> 13) & 0x3F)
+
+typedef struct {
+    Vec3f scale;
+    Vec3s rot;
+    Vec3f pos;
+} ScaleRotPos;
+
+typedef struct {
+    /* 0x00 */ u16 type;
+    union {
+        u16 vtxData[3];
+        struct {
+            /* 0x02 */ u16 flags_vIA; // 0xE000 is poly exclusion flags, 0x1FFF is vtxId
+            /* 0x04 */ u16 flags_vIB; // 0xE000 is ? flags, 0x1FFF is vtxId
+            /* 0x06 */ u16 vIC;
+        };
+    };
+    /* 0x08 */ Vec3s norm; // Unit normal vector.
+                           // Value ranges from -0x7FFF to 0x7FFF, representing -1 to 1. 0x8000 is invalid
+
+    /* 0x0E */ s16 dist; // Plane distance from origin
+} CollisionPoly;         // size = 0x10
+
+typedef struct {
+    /* 0x00 */ u16 cameraSType;
+    /* 0x02 */ s16 numCameras;
+    /* 0x04 */ Vec3s* camPosData;
+} CamData;
+
+typedef struct {
+    /* 0x00 */ s16 xMin;
+    /* 0x02 */ s16 ySurface;
+    /* 0x04 */ s16 zMin;
+    /* 0x06 */ s16 xLength;
+    /* 0x08 */ s16 zLength;
+    /* 0x0C */ u32 properties;
+
+    // 0x0008_0000 = ?
+    // 0x0007_E000 = Room Index, 0x3F = all rooms
+    // 0x0000_1F00 = Lighting Settings Index
+    // 0x0000_00FF = CamData index
+} WaterBox; // size = 0x10
+
+typedef struct {
+    u32 data[2];
+} SurfaceType;
+// Type 1
+// 0x0800_0000 = wall damage
+
+typedef struct {
+    /* 0x00 */ Vec3s minBounds; // colAbsMin
+    /* 0x06 */ Vec3s maxBounds; // colAbsMax
+    /* 0x0C */ u16 nbVertices;
+    /* 0x10 */ Vec3s* vtxList; // vertexArray
+    /* 0x14 */ u16 nbPolygons;
+    /* 0x18 */ CollisionPoly* polyList; // polygonArray
+    /* 0x1C */ SurfaceType* polygonTypes;
+    /* 0x20 */ CamData* cameraDataList;
+    /* 0x24 */ u16 nbWaterBoxes;
+    /* 0x28 */ WaterBox* waterBoxes;
+} CollisionHeader; // BGDataInfo
+
+typedef struct {
+    s16 polyId;
+    u16 next; // next SSNode index
+} SSNode;
```

Comment:
what is SS? this comes from the original name I think, but I think if it could have a better more descriptive name then it should. Same goes for anything else with SS in it

Reply:
short_slist_node